Feature/config

spellgen opened the pull request:
See what you think about style and naming

Commit:
cleaning up tests

Commit:
adding service config

Commit:
service and prefix, documentation

Commit:
in working condition

aaronhurt commented on configurator_test.go:
```diff
@@ -62,14 +104,22 @@ func (c *config) Update(_ uint64, data interface{}) {
+func (c *config) copy() *config {
+	c.RLock()
+	defer c.RUnlock()
+	cc := *c
```
This is failing govet...

```

aaronhurt replied:
This is failing govet in the travis build: https://travis-ci.org/myENA/consultant/jobs/236450500

```
configurator_test.go:119: assignment copies lock value to cc: consultant_test.config
```

Commit:
now completely different

spellgen commented:
doing this the go way instead - chan chan chan all the way

Commit:
takenap(1000)

dcarbone committed:
Few small changes:

- `go fmt`'d the files
- Renamed test constants with "configurator" prefix to avoid future
  conflicts
- Changed uses of "fmt.Print*" to "log.Print*" to take advantage of
  package logger

dcarbone commented:
lgtm :)
